### PR TITLE
ClassMethod: add checking on magic method

### PR DESCRIPTION
**Pull request by DragorWW**
code for #378 issue

changes:
- ClassMethod: add method isMagic
- ClassMethodTest: add test for isMagic

_All is well, or need to change?_

### CODE REVIEW

File: lib/PhpParser/Node/Stmt/ClassMethod.php [hunk 1]
```diff
@@ -117,4 +135,13 @@ public function isFinal() {
...
         return (bool) ($this->flags & Class_::MODIFIER_STATIC);
     }
+
+    /**
+     * Whether the method is magic.
+     *
+     * @return bool
+     */
+    public function isMagic() {
+        return (bool) in_array($this->name, static::MAGIC_NAMES);
```
Comment (nikic) [inline lib/PhpParser/Node/Stmt/ClassMethod.php]: Method names in PHP are case-insensitive, so this should do a `strtolower` call first.

Comment (nikic) [inline lib/PhpParser/Node/Stmt/ClassMethod.php]: Additionally, I would suggest to convert `MAGIC_NAMES` into a map (see for example https://github.com/DragorWW/PHP-Parser/blob/00d3ca3ccdeb14480a55af6c47d1ca9bd5e9cde7/lib/PhpParser/Node/Stmt/Class_.php#L28) and then check whether it is a magic method using `isset()`. This is more efficient than using in_array.

Comment (DragorWW) [inline lib/PhpParser/Node/Stmt/ClassMethod.php]: @nikic may be `const MAGIC_NAMES` change to `protected static $magicNames`?

Comment (nikic) [inline lib/PhpParser/Node/Stmt/ClassMethod.php]: Yeah, looks like the const array currently breaks the build on 5.6 and on HHVM.

Comment (DragorWW) [inline lib/PhpParser/Node/Stmt/ClassMethod.php]: @nikic fix it

**Comment by nikic** [comment]
Merged as https://github.com/nikic/PHP-Parser/commit/9dc93aafaab90eb885a59412237b323047c1819a. Thanks!

**Pull request closed without merging**